pallets/settlement/src/lib.rs

```diff
@@ -711,9 +597,7 @@ decl_module! {
         /// # Arguments
         /// * `instruction_id` - Instruction id to affirm.
         /// * `portfolios` - Portfolios that the sender controls and wants to affirm this instruction
-        #[weight = weight_for::weight_for_affirmation_instruction::<T>()
-            + weight_for::weight_for_transfer::<T>() // Maximum weight for `execute_instruction()`
-        ]
+        #[weight = <T as Trait>::WeightInfo::affirm_instruction(portfolios.len() as u32)]
```

Comment:
How are you limiting legs here? One portfolio can be involved in multiple legs and the weight should be a function of the number of legs, not the number of portfolios.

Reply:
It is not recommended to read the storage during weight calculation. So what I did is to assume that every leg needs a unique portfolio and portfolio length is always equal to no. of legs for a given instruction_id. But I think it will give us an incorrect weight when we don't have that assumption true. Another approach we can follow is to always cost for maximum no. of legs but this will affect too much on the cost of the dispatchable. 

We can do a compromise something like this decrease the no. of legs in instruction and always charge for the max wherever the portfolios come into the picture.

For now I took the easy assumption rest we can ask the product feedback.

Reply:
We can not work on this assumption. A single portfolio can trigger the max limit of legs. We have two main ways to solve this
1) Ask the user to input the number of legs
2) Always charge for max legs

I think it makes sense to use the first approach. SDK can abstract this from the user.

Reply:
Then It is a breaking API change. We need product feedback here. I think it is good to have a separate story for this.

Reply:
It is but there's no other way around it. We can discuss over the standup. I don't think we need a separate story for this.

Reply:
I'm comfortable with either approach (legs as a param, charging for max legs) - I wonder if we should go down the latter approach for simplicity for now and to avoid API changes given that this will be turned off on go-live?